Content.Server/Explosions/ExplosionHelper.cs

```diff
@@ -76,29 +77,33 @@ public static void SpawnExplosion(GridCoordinates coords, int devastationRange,
             foreach (var tile in tiles)
             {
                 var tileLoc = mapGrid.GridTileToLocal(tile.GridIndices);
-                var tileDef = (ContentTileDefinition)tileDefinitionManager[tile.Tile.TypeId];
-                var distanceFromTile = (int)tileLoc.Distance(mapManager, coords);
-                if (!string.IsNullOrWhiteSpace(tileDef.SubFloor))
+                var tileDef = (ContentTileDefinition) tileDefinitionManager[tile.Tile.TypeId];
+                ContentTileDefinition.BaseTurfs.TryGetValue(tileDef.Name, out var baseTurfs);
+                var distanceFromTile = (int) tileLoc.Distance(mapManager, coords);
+                if (!baseTurfs.Any())
+                {
+                    continue;
+                }
+                if (distanceFromTile < devastationRange)
                 {
-                    if (distanceFromTile < devastationRange)
-                        mapGrid.SetTile(tileLoc, new Tile(tileDefinitionManager["space"].TileId));
-                    if (distanceFromTile < heavyImpactRange)
+                    mapGrid.SetTile(tileLoc, new Tile(tileDefinitionManager[baseTurfs[0]].TileId));
+                }
+                else if (distanceFromTile < heavyImpactRange)
+                {
+                    if (robustRandom.Prob(80))
+                    {
+                        mapGrid.SetTile(tileLoc, new Tile(tileDefinitionManager[baseTurfs[baseTurfs.Count - 1]].TileId));
```

Comment:
```suggestion
                        mapGrid.SetTile(tileLoc, new Tile(tileDefinitionManager[baseTurfs[^1]].TileId));
```

You can use C# 8 `Index` here.